From 22411ac4441750bf464a37e39ef85e76fed13e14 Mon Sep 17 00:00:00 2001
From: Szabolcs Nagy <szabolcs.nagy@arm.com>
Date: Fri, 10 Nov 2017 14:15:13 +0000
Subject: [PATCH 07/10] aarch64: Update ILP32 abilists

Update libc and libm abilists with memfd_create and new _FloatN symbols.
Remove __nis_hash and __p_secstodate from libnsl and libresolv abilists.
This will need to be merged with the initial ilp32 abilist commit.

2017-11-27  Szabolcs Nagy  <szabolcs.nagy@arm.com>

	* sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist: Update.
	* sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist: Update.
	* sysdeps/unix/sysv/linux/aarch64/ilp32/libnsl.abilist: Update.
	* sysdeps/unix/sysv/linux/aarch64/ilp32/libresolv.abilist: Update.
---
 sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist |  11 ++
 sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist | 208 +++++++++++++++++++++
 .../unix/sysv/linux/aarch64/ilp32/libnsl.abilist   |   1 -
 .../sysv/linux/aarch64/ilp32/libresolv.abilist     |   1 -
 4 files changed, 219 insertions(+), 2 deletions(-)

diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
index fa390087a0..dd0ed329ab 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libc.abilist
@@ -1247,6 +1247,7 @@ GLIBC_2.27 memccpy F
 GLIBC_2.27 memchr F
 GLIBC_2.27 memcmp F
 GLIBC_2.27 memcpy F
+GLIBC_2.27 memfd_create F
 GLIBC_2.27 memfrob F
 GLIBC_2.27 memmem F
 GLIBC_2.27 memmove F
@@ -1710,6 +1711,8 @@ GLIBC_2.27 strfmon F
 GLIBC_2.27 strfmon_l F
 GLIBC_2.27 strfromd F
 GLIBC_2.27 strfromf F
+GLIBC_2.27 strfromf128 F
+GLIBC_2.27 strfromf64x F
 GLIBC_2.27 strfroml F
 GLIBC_2.27 strfry F
 GLIBC_2.27 strftime F
@@ -1733,6 +1736,10 @@ GLIBC_2.27 strstr F
 GLIBC_2.27 strtod F
 GLIBC_2.27 strtod_l F
 GLIBC_2.27 strtof F
+GLIBC_2.27 strtof128 F
+GLIBC_2.27 strtof128_l F
+GLIBC_2.27 strtof64x F
+GLIBC_2.27 strtof64x_l F
 GLIBC_2.27 strtof_l F
 GLIBC_2.27 strtoimax F
 GLIBC_2.27 strtok F
@@ -1946,6 +1953,10 @@ GLIBC_2.27 wcsstr F
 GLIBC_2.27 wcstod F
 GLIBC_2.27 wcstod_l F
 GLIBC_2.27 wcstof F
+GLIBC_2.27 wcstof128 F
+GLIBC_2.27 wcstof128_l F
+GLIBC_2.27 wcstof64x F
+GLIBC_2.27 wcstof64x_l F
 GLIBC_2.27 wcstof_l F
 GLIBC_2.27 wcstoimax F
 GLIBC_2.27 wcstok F
diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
index c1decebe49..51a53dafd1 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libm.abilist
@@ -101,110 +101,182 @@ GLIBC_2.27 __ynf_finite F
 GLIBC_2.27 __ynl_finite F
 GLIBC_2.27 acos F
 GLIBC_2.27 acosf F
+GLIBC_2.27 acosf128 F
+GLIBC_2.27 acosf64x F
 GLIBC_2.27 acosh F
 GLIBC_2.27 acoshf F
+GLIBC_2.27 acoshf128 F
+GLIBC_2.27 acoshf64x F
 GLIBC_2.27 acoshl F
 GLIBC_2.27 acosl F
 GLIBC_2.27 asin F
 GLIBC_2.27 asinf F
+GLIBC_2.27 asinf128 F
+GLIBC_2.27 asinf64x F
 GLIBC_2.27 asinh F
 GLIBC_2.27 asinhf F
+GLIBC_2.27 asinhf128 F
+GLIBC_2.27 asinhf64x F
 GLIBC_2.27 asinhl F
 GLIBC_2.27 asinl F
 GLIBC_2.27 atan F
 GLIBC_2.27 atan2 F
 GLIBC_2.27 atan2f F
+GLIBC_2.27 atan2f128 F
+GLIBC_2.27 atan2f64x F
 GLIBC_2.27 atan2l F
 GLIBC_2.27 atanf F
+GLIBC_2.27 atanf128 F
+GLIBC_2.27 atanf64x F
 GLIBC_2.27 atanh F
 GLIBC_2.27 atanhf F
+GLIBC_2.27 atanhf128 F
+GLIBC_2.27 atanhf64x F
 GLIBC_2.27 atanhl F
 GLIBC_2.27 atanl F
 GLIBC_2.27 cabs F
 GLIBC_2.27 cabsf F
+GLIBC_2.27 cabsf128 F
+GLIBC_2.27 cabsf64x F
 GLIBC_2.27 cabsl F
 GLIBC_2.27 cacos F
 GLIBC_2.27 cacosf F
+GLIBC_2.27 cacosf128 F
+GLIBC_2.27 cacosf64x F
 GLIBC_2.27 cacosh F
 GLIBC_2.27 cacoshf F
+GLIBC_2.27 cacoshf128 F
+GLIBC_2.27 cacoshf64x F
 GLIBC_2.27 cacoshl F
 GLIBC_2.27 cacosl F
 GLIBC_2.27 canonicalize F
 GLIBC_2.27 canonicalizef F
+GLIBC_2.27 canonicalizef128 F
+GLIBC_2.27 canonicalizef64x F
 GLIBC_2.27 canonicalizel F
 GLIBC_2.27 carg F
 GLIBC_2.27 cargf F
+GLIBC_2.27 cargf128 F
+GLIBC_2.27 cargf64x F
 GLIBC_2.27 cargl F
 GLIBC_2.27 casin F
 GLIBC_2.27 casinf F
+GLIBC_2.27 casinf128 F
+GLIBC_2.27 casinf64x F
 GLIBC_2.27 casinh F
 GLIBC_2.27 casinhf F
+GLIBC_2.27 casinhf128 F
+GLIBC_2.27 casinhf64x F
 GLIBC_2.27 casinhl F
 GLIBC_2.27 casinl F
 GLIBC_2.27 catan F
 GLIBC_2.27 catanf F
+GLIBC_2.27 catanf128 F
+GLIBC_2.27 catanf64x F
 GLIBC_2.27 catanh F
 GLIBC_2.27 catanhf F
+GLIBC_2.27 catanhf128 F
+GLIBC_2.27 catanhf64x F
 GLIBC_2.27 catanhl F
 GLIBC_2.27 catanl F
 GLIBC_2.27 cbrt F
 GLIBC_2.27 cbrtf F
+GLIBC_2.27 cbrtf128 F
+GLIBC_2.27 cbrtf64x F
 GLIBC_2.27 cbrtl F
 GLIBC_2.27 ccos F
 GLIBC_2.27 ccosf F
+GLIBC_2.27 ccosf128 F
+GLIBC_2.27 ccosf64x F
 GLIBC_2.27 ccosh F
 GLIBC_2.27 ccoshf F
+GLIBC_2.27 ccoshf128 F
+GLIBC_2.27 ccoshf64x F
 GLIBC_2.27 ccoshl F
 GLIBC_2.27 ccosl F
 GLIBC_2.27 ceil F
 GLIBC_2.27 ceilf F
+GLIBC_2.27 ceilf128 F
+GLIBC_2.27 ceilf64x F
 GLIBC_2.27 ceill F
 GLIBC_2.27 cexp F
 GLIBC_2.27 cexpf F
+GLIBC_2.27 cexpf128 F
+GLIBC_2.27 cexpf64x F
 GLIBC_2.27 cexpl F
 GLIBC_2.27 cimag F
 GLIBC_2.27 cimagf F
+GLIBC_2.27 cimagf128 F
+GLIBC_2.27 cimagf64x F
 GLIBC_2.27 cimagl F
 GLIBC_2.27 clog F
 GLIBC_2.27 clog10 F
 GLIBC_2.27 clog10f F
+GLIBC_2.27 clog10f128 F
+GLIBC_2.27 clog10f64x F
 GLIBC_2.27 clog10l F
 GLIBC_2.27 clogf F
+GLIBC_2.27 clogf128 F
+GLIBC_2.27 clogf64x F
 GLIBC_2.27 clogl F
 GLIBC_2.27 conj F
 GLIBC_2.27 conjf F
+GLIBC_2.27 conjf128 F
+GLIBC_2.27 conjf64x F
 GLIBC_2.27 conjl F
 GLIBC_2.27 copysign F
 GLIBC_2.27 copysignf F
+GLIBC_2.27 copysignf128 F
+GLIBC_2.27 copysignf64x F
 GLIBC_2.27 copysignl F
 GLIBC_2.27 cos F
 GLIBC_2.27 cosf F
+GLIBC_2.27 cosf128 F
+GLIBC_2.27 cosf64x F
 GLIBC_2.27 cosh F
 GLIBC_2.27 coshf F
+GLIBC_2.27 coshf128 F
+GLIBC_2.27 coshf64x F
 GLIBC_2.27 coshl F
 GLIBC_2.27 cosl F
 GLIBC_2.27 cpow F
 GLIBC_2.27 cpowf F
+GLIBC_2.27 cpowf128 F
+GLIBC_2.27 cpowf64x F
 GLIBC_2.27 cpowl F
 GLIBC_2.27 cproj F
 GLIBC_2.27 cprojf F
+GLIBC_2.27 cprojf128 F
+GLIBC_2.27 cprojf64x F
 GLIBC_2.27 cprojl F
 GLIBC_2.27 creal F
 GLIBC_2.27 crealf F
+GLIBC_2.27 crealf128 F
+GLIBC_2.27 crealf64x F
 GLIBC_2.27 creall F
 GLIBC_2.27 csin F
 GLIBC_2.27 csinf F
+GLIBC_2.27 csinf128 F
+GLIBC_2.27 csinf64x F
 GLIBC_2.27 csinh F
 GLIBC_2.27 csinhf F
+GLIBC_2.27 csinhf128 F
+GLIBC_2.27 csinhf64x F
 GLIBC_2.27 csinhl F
 GLIBC_2.27 csinl F
 GLIBC_2.27 csqrt F
 GLIBC_2.27 csqrtf F
+GLIBC_2.27 csqrtf128 F
+GLIBC_2.27 csqrtf64x F
 GLIBC_2.27 csqrtl F
 GLIBC_2.27 ctan F
 GLIBC_2.27 ctanf F
+GLIBC_2.27 ctanf128 F
+GLIBC_2.27 ctanf64x F
 GLIBC_2.27 ctanh F
 GLIBC_2.27 ctanhf F
+GLIBC_2.27 ctanhf128 F
+GLIBC_2.27 ctanhf64x F
 GLIBC_2.27 ctanhl F
 GLIBC_2.27 ctanl F
 GLIBC_2.27 drem F
@@ -213,26 +285,42 @@ GLIBC_2.27 dreml F
 GLIBC_2.27 erf F
 GLIBC_2.27 erfc F
 GLIBC_2.27 erfcf F
+GLIBC_2.27 erfcf128 F
+GLIBC_2.27 erfcf64x F
 GLIBC_2.27 erfcl F
 GLIBC_2.27 erff F
+GLIBC_2.27 erff128 F
+GLIBC_2.27 erff64x F
 GLIBC_2.27 erfl F
 GLIBC_2.27 exp F
 GLIBC_2.27 exp10 F
 GLIBC_2.27 exp10f F
+GLIBC_2.27 exp10f128 F
+GLIBC_2.27 exp10f64x F
 GLIBC_2.27 exp10l F
 GLIBC_2.27 exp2 F
 GLIBC_2.27 exp2f F
+GLIBC_2.27 exp2f128 F
+GLIBC_2.27 exp2f64x F
 GLIBC_2.27 exp2l F
 GLIBC_2.27 expf F
+GLIBC_2.27 expf128 F
+GLIBC_2.27 expf64x F
 GLIBC_2.27 expl F
 GLIBC_2.27 expm1 F
 GLIBC_2.27 expm1f F
+GLIBC_2.27 expm1f128 F
+GLIBC_2.27 expm1f64x F
 GLIBC_2.27 expm1l F
 GLIBC_2.27 fabs F
 GLIBC_2.27 fabsf F
+GLIBC_2.27 fabsf128 F
+GLIBC_2.27 fabsf64x F
 GLIBC_2.27 fabsl F
 GLIBC_2.27 fdim F
 GLIBC_2.27 fdimf F
+GLIBC_2.27 fdimf128 F
+GLIBC_2.27 fdimf64x F
 GLIBC_2.27 fdiml F
 GLIBC_2.27 feclearexcept F
 GLIBC_2.27 fedisableexcept F
@@ -257,147 +345,237 @@ GLIBC_2.27 finitef F
 GLIBC_2.27 finitel F
 GLIBC_2.27 floor F
 GLIBC_2.27 floorf F
+GLIBC_2.27 floorf128 F
+GLIBC_2.27 floorf64x F
 GLIBC_2.27 floorl F
 GLIBC_2.27 fma F
 GLIBC_2.27 fmaf F
+GLIBC_2.27 fmaf128 F
+GLIBC_2.27 fmaf64x F
 GLIBC_2.27 fmal F
 GLIBC_2.27 fmax F
 GLIBC_2.27 fmaxf F
+GLIBC_2.27 fmaxf128 F
+GLIBC_2.27 fmaxf64x F
 GLIBC_2.27 fmaxl F
 GLIBC_2.27 fmaxmag F
 GLIBC_2.27 fmaxmagf F
+GLIBC_2.27 fmaxmagf128 F
+GLIBC_2.27 fmaxmagf64x F
 GLIBC_2.27 fmaxmagl F
 GLIBC_2.27 fmin F
 GLIBC_2.27 fminf F
+GLIBC_2.27 fminf128 F
+GLIBC_2.27 fminf64x F
 GLIBC_2.27 fminl F
 GLIBC_2.27 fminmag F
 GLIBC_2.27 fminmagf F
+GLIBC_2.27 fminmagf128 F
+GLIBC_2.27 fminmagf64x F
 GLIBC_2.27 fminmagl F
 GLIBC_2.27 fmod F
 GLIBC_2.27 fmodf F
+GLIBC_2.27 fmodf128 F
+GLIBC_2.27 fmodf64x F
 GLIBC_2.27 fmodl F
 GLIBC_2.27 frexp F
 GLIBC_2.27 frexpf F
+GLIBC_2.27 frexpf128 F
+GLIBC_2.27 frexpf64x F
 GLIBC_2.27 frexpl F
 GLIBC_2.27 fromfp F
 GLIBC_2.27 fromfpf F
+GLIBC_2.27 fromfpf128 F
+GLIBC_2.27 fromfpf64x F
 GLIBC_2.27 fromfpl F
 GLIBC_2.27 fromfpx F
 GLIBC_2.27 fromfpxf F
+GLIBC_2.27 fromfpxf128 F
+GLIBC_2.27 fromfpxf64x F
 GLIBC_2.27 fromfpxl F
 GLIBC_2.27 gamma F
 GLIBC_2.27 gammaf F
 GLIBC_2.27 gammal F
 GLIBC_2.27 getpayload F
 GLIBC_2.27 getpayloadf F
+GLIBC_2.27 getpayloadf128 F
+GLIBC_2.27 getpayloadf64x F
 GLIBC_2.27 getpayloadl F
 GLIBC_2.27 hypot F
 GLIBC_2.27 hypotf F
+GLIBC_2.27 hypotf128 F
+GLIBC_2.27 hypotf64x F
 GLIBC_2.27 hypotl F
 GLIBC_2.27 ilogb F
 GLIBC_2.27 ilogbf F
+GLIBC_2.27 ilogbf128 F
+GLIBC_2.27 ilogbf64x F
 GLIBC_2.27 ilogbl F
 GLIBC_2.27 j0 F
 GLIBC_2.27 j0f F
+GLIBC_2.27 j0f128 F
+GLIBC_2.27 j0f64x F
 GLIBC_2.27 j0l F
 GLIBC_2.27 j1 F
 GLIBC_2.27 j1f F
+GLIBC_2.27 j1f128 F
+GLIBC_2.27 j1f64x F
 GLIBC_2.27 j1l F
 GLIBC_2.27 jn F
 GLIBC_2.27 jnf F
+GLIBC_2.27 jnf128 F
+GLIBC_2.27 jnf64x F
 GLIBC_2.27 jnl F
 GLIBC_2.27 ldexp F
 GLIBC_2.27 ldexpf F
+GLIBC_2.27 ldexpf128 F
+GLIBC_2.27 ldexpf64x F
 GLIBC_2.27 ldexpl F
 GLIBC_2.27 lgamma F
 GLIBC_2.27 lgamma_r F
 GLIBC_2.27 lgammaf F
+GLIBC_2.27 lgammaf128 F
+GLIBC_2.27 lgammaf128_r F
+GLIBC_2.27 lgammaf64x F
+GLIBC_2.27 lgammaf64x_r F
 GLIBC_2.27 lgammaf_r F
 GLIBC_2.27 lgammal F
 GLIBC_2.27 lgammal_r F
 GLIBC_2.27 llogb F
 GLIBC_2.27 llogbf F
+GLIBC_2.27 llogbf128 F
+GLIBC_2.27 llogbf64x F
 GLIBC_2.27 llogbl F
 GLIBC_2.27 llrint F
 GLIBC_2.27 llrintf F
+GLIBC_2.27 llrintf128 F
+GLIBC_2.27 llrintf64x F
 GLIBC_2.27 llrintl F
 GLIBC_2.27 llround F
 GLIBC_2.27 llroundf F
+GLIBC_2.27 llroundf128 F
+GLIBC_2.27 llroundf64x F
 GLIBC_2.27 llroundl F
 GLIBC_2.27 log F
 GLIBC_2.27 log10 F
 GLIBC_2.27 log10f F
+GLIBC_2.27 log10f128 F
+GLIBC_2.27 log10f64x F
 GLIBC_2.27 log10l F
 GLIBC_2.27 log1p F
 GLIBC_2.27 log1pf F
+GLIBC_2.27 log1pf128 F
+GLIBC_2.27 log1pf64x F
 GLIBC_2.27 log1pl F
 GLIBC_2.27 log2 F
 GLIBC_2.27 log2f F
+GLIBC_2.27 log2f128 F
+GLIBC_2.27 log2f64x F
 GLIBC_2.27 log2l F
 GLIBC_2.27 logb F
 GLIBC_2.27 logbf F
+GLIBC_2.27 logbf128 F
+GLIBC_2.27 logbf64x F
 GLIBC_2.27 logbl F
 GLIBC_2.27 logf F
+GLIBC_2.27 logf128 F
+GLIBC_2.27 logf64x F
 GLIBC_2.27 logl F
 GLIBC_2.27 lrint F
 GLIBC_2.27 lrintf F
+GLIBC_2.27 lrintf128 F
+GLIBC_2.27 lrintf64x F
 GLIBC_2.27 lrintl F
 GLIBC_2.27 lround F
 GLIBC_2.27 lroundf F
+GLIBC_2.27 lroundf128 F
+GLIBC_2.27 lroundf64x F
 GLIBC_2.27 lroundl F
 GLIBC_2.27 modf F
 GLIBC_2.27 modff F
+GLIBC_2.27 modff128 F
+GLIBC_2.27 modff64x F
 GLIBC_2.27 modfl F
 GLIBC_2.27 nan F
 GLIBC_2.27 nanf F
+GLIBC_2.27 nanf128 F
+GLIBC_2.27 nanf64x F
 GLIBC_2.27 nanl F
 GLIBC_2.27 nearbyint F
 GLIBC_2.27 nearbyintf F
+GLIBC_2.27 nearbyintf128 F
+GLIBC_2.27 nearbyintf64x F
 GLIBC_2.27 nearbyintl F
 GLIBC_2.27 nextafter F
 GLIBC_2.27 nextafterf F
+GLIBC_2.27 nextafterf128 F
+GLIBC_2.27 nextafterf64x F
 GLIBC_2.27 nextafterl F
 GLIBC_2.27 nextdown F
 GLIBC_2.27 nextdownf F
+GLIBC_2.27 nextdownf128 F
+GLIBC_2.27 nextdownf64x F
 GLIBC_2.27 nextdownl F
 GLIBC_2.27 nexttoward F
 GLIBC_2.27 nexttowardf F
 GLIBC_2.27 nexttowardl F
 GLIBC_2.27 nextup F
 GLIBC_2.27 nextupf F
+GLIBC_2.27 nextupf128 F
+GLIBC_2.27 nextupf64x F
 GLIBC_2.27 nextupl F
 GLIBC_2.27 pow F
 GLIBC_2.27 powf F
+GLIBC_2.27 powf128 F
+GLIBC_2.27 powf64x F
 GLIBC_2.27 powl F
 GLIBC_2.27 remainder F
 GLIBC_2.27 remainderf F
+GLIBC_2.27 remainderf128 F
+GLIBC_2.27 remainderf64x F
 GLIBC_2.27 remainderl F
 GLIBC_2.27 remquo F
 GLIBC_2.27 remquof F
+GLIBC_2.27 remquof128 F
+GLIBC_2.27 remquof64x F
 GLIBC_2.27 remquol F
 GLIBC_2.27 rint F
 GLIBC_2.27 rintf F
+GLIBC_2.27 rintf128 F
+GLIBC_2.27 rintf64x F
 GLIBC_2.27 rintl F
 GLIBC_2.27 round F
 GLIBC_2.27 roundeven F
 GLIBC_2.27 roundevenf F
+GLIBC_2.27 roundevenf128 F
+GLIBC_2.27 roundevenf64x F
 GLIBC_2.27 roundevenl F
 GLIBC_2.27 roundf F
+GLIBC_2.27 roundf128 F
+GLIBC_2.27 roundf64x F
 GLIBC_2.27 roundl F
 GLIBC_2.27 scalb F
 GLIBC_2.27 scalbf F
 GLIBC_2.27 scalbl F
 GLIBC_2.27 scalbln F
 GLIBC_2.27 scalblnf F
+GLIBC_2.27 scalblnf128 F
+GLIBC_2.27 scalblnf64x F
 GLIBC_2.27 scalblnl F
 GLIBC_2.27 scalbn F
 GLIBC_2.27 scalbnf F
+GLIBC_2.27 scalbnf128 F
+GLIBC_2.27 scalbnf64x F
 GLIBC_2.27 scalbnl F
 GLIBC_2.27 setpayload F
 GLIBC_2.27 setpayloadf F
+GLIBC_2.27 setpayloadf128 F
+GLIBC_2.27 setpayloadf64x F
 GLIBC_2.27 setpayloadl F
 GLIBC_2.27 setpayloadsig F
 GLIBC_2.27 setpayloadsigf F
+GLIBC_2.27 setpayloadsigf128 F
+GLIBC_2.27 setpayloadsigf64x F
 GLIBC_2.27 setpayloadsigl F
 GLIBC_2.27 signgam D 0x4
 GLIBC_2.27 significand F
@@ -406,45 +584,75 @@ GLIBC_2.27 significandl F
 GLIBC_2.27 sin F
 GLIBC_2.27 sincos F
 GLIBC_2.27 sincosf F
+GLIBC_2.27 sincosf128 F
+GLIBC_2.27 sincosf64x F
 GLIBC_2.27 sincosl F
 GLIBC_2.27 sinf F
+GLIBC_2.27 sinf128 F
+GLIBC_2.27 sinf64x F
 GLIBC_2.27 sinh F
 GLIBC_2.27 sinhf F
+GLIBC_2.27 sinhf128 F
+GLIBC_2.27 sinhf64x F
 GLIBC_2.27 sinhl F
 GLIBC_2.27 sinl F
 GLIBC_2.27 sqrt F
 GLIBC_2.27 sqrtf F
+GLIBC_2.27 sqrtf128 F
+GLIBC_2.27 sqrtf64x F
 GLIBC_2.27 sqrtl F
 GLIBC_2.27 tan F
 GLIBC_2.27 tanf F
+GLIBC_2.27 tanf128 F
+GLIBC_2.27 tanf64x F
 GLIBC_2.27 tanh F
 GLIBC_2.27 tanhf F
+GLIBC_2.27 tanhf128 F
+GLIBC_2.27 tanhf64x F
 GLIBC_2.27 tanhl F
 GLIBC_2.27 tanl F
 GLIBC_2.27 tgamma F
 GLIBC_2.27 tgammaf F
+GLIBC_2.27 tgammaf128 F
+GLIBC_2.27 tgammaf64x F
 GLIBC_2.27 tgammal F
 GLIBC_2.27 totalorder F
 GLIBC_2.27 totalorderf F
+GLIBC_2.27 totalorderf128 F
+GLIBC_2.27 totalorderf64x F
 GLIBC_2.27 totalorderl F
 GLIBC_2.27 totalordermag F
 GLIBC_2.27 totalordermagf F
+GLIBC_2.27 totalordermagf128 F
+GLIBC_2.27 totalordermagf64x F
 GLIBC_2.27 totalordermagl F
 GLIBC_2.27 trunc F
 GLIBC_2.27 truncf F
+GLIBC_2.27 truncf128 F
+GLIBC_2.27 truncf64x F
 GLIBC_2.27 truncl F
 GLIBC_2.27 ufromfp F
 GLIBC_2.27 ufromfpf F
+GLIBC_2.27 ufromfpf128 F
+GLIBC_2.27 ufromfpf64x F
 GLIBC_2.27 ufromfpl F
 GLIBC_2.27 ufromfpx F
 GLIBC_2.27 ufromfpxf F
+GLIBC_2.27 ufromfpxf128 F
+GLIBC_2.27 ufromfpxf64x F
 GLIBC_2.27 ufromfpxl F
 GLIBC_2.27 y0 F
 GLIBC_2.27 y0f F
+GLIBC_2.27 y0f128 F
+GLIBC_2.27 y0f64x F
 GLIBC_2.27 y0l F
 GLIBC_2.27 y1 F
 GLIBC_2.27 y1f F
+GLIBC_2.27 y1f128 F
+GLIBC_2.27 y1f64x F
 GLIBC_2.27 y1l F
 GLIBC_2.27 yn F
 GLIBC_2.27 ynf F
+GLIBC_2.27 ynf128 F
+GLIBC_2.27 ynf64x F
 GLIBC_2.27 ynl F
diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libnsl.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libnsl.abilist
index 1f6ec0ba9b..b7a0f82a9e 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libnsl.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libnsl.abilist
@@ -5,7 +5,6 @@ GLIBC_2.27 __nis_default_group F
 GLIBC_2.27 __nis_default_owner F
 GLIBC_2.27 __nis_default_ttl F
 GLIBC_2.27 __nis_finddirectory F
-GLIBC_2.27 __nis_hash F
 GLIBC_2.27 __nisbind_connect F
 GLIBC_2.27 __nisbind_create F
 GLIBC_2.27 __nisbind_destroy F
diff --git a/sysdeps/unix/sysv/linux/aarch64/ilp32/libresolv.abilist b/sysdeps/unix/sysv/linux/aarch64/ilp32/libresolv.abilist
index cf1374f1e4..42af2652a7 100644
--- a/sysdeps/unix/sysv/linux/aarch64/ilp32/libresolv.abilist
+++ b/sysdeps/unix/sysv/linux/aarch64/ilp32/libresolv.abilist
@@ -20,7 +20,6 @@ GLIBC_2.27 __p_fqnname F
 GLIBC_2.27 __p_option F
 GLIBC_2.27 __p_query F
 GLIBC_2.27 __p_rcode F
-GLIBC_2.27 __p_secstodate F
 GLIBC_2.27 __p_time F
 GLIBC_2.27 __p_type F
 GLIBC_2.27 __p_type_syms D 0x228
-- 
2.11.1

